src/Promitor.Agents.Scraper/Scheduling/SchedulingExtensions.cs

```diff
@@ -33,58 +29,86 @@ public static IServiceCollection ScheduleMetricScraping(this IServiceCollection
+        private static void ScheduleResourcesScraping(MetricsDeclaration metricsDeclaration,
+            MetricSinkWriter metricSinkWriter,
+            AzureMonitorClientFactory azureMonitorClientFactory,
+            IAzureScrapingPrometheusMetricsCollector azureScrapingPrometheusMetricsCollector,
+            IMemoryCache resourceMetricDefinitionMemoryCache,
+            IScrapingMutex scrapingTaskMutex,
+            IConfiguration configuration,
+            IOptions<AzureMonitorLoggingConfiguration> azureMonitorLoggingConfiguration,
+            ILoggerFactory loggerFactory,
+            ILogger<Startup> logger,
+            IServiceCollection services)
         {
-            var resourceSubscriptionId = string.IsNullOrWhiteSpace(resource.SubscriptionId) ? azureMetadata.SubscriptionId : resource.SubscriptionId;
-            var azureMonitorClient = azureMonitorClientFactory.CreateIfNotExists(azureMetadata.Cloud, azureMetadata.TenantId, resourceSubscriptionId, metricSinkWriter, azureScrapingPrometheusMetricCollector, memoryCache, configuration, azureMonitorLoggingConfiguration, loggerFactory);
-            var scrapeDefinition = metric.CreateScrapeDefinition(resource, azureMetadata);
-            var jobName = GenerateResourceScrapingJobName(scrapeDefinition, resource);
-
+            var jobName = GenerateResourceScrapingJobName(metricsDeclaration, logger);
+            
             services.AddScheduler(builder =>
             {
                 builder.AddJob(jobServices =>
-                    {
-                        return new ResourceScrapingJob(jobName, scrapeDefinition,
+                        new ResourcesScrapingJob(jobName,
+                            metricsDeclaration,
+                            jobServices.GetService<ResourceDiscoveryRepository>(),
                             metricSinkWriter,
                             jobServices.GetService<MetricScraperFactory>(),
-                            azureMonitorClient,
-                            jobServices.GetService<IAzureScrapingPrometheusMetricsCollector>(),
-                            jobServices.GetService<ILogger<ResourceScrapingJob>>());
-                    }, schedulerOptions =>
+                            azureMonitorClientFactory,
+                            azureScrapingPrometheusMetricsCollector,
+                            resourceMetricDefinitionMemoryCache,
+                            scrapingTaskMutex,
+                            configuration,
+                            azureMonitorLoggingConfiguration,
+                            loggerFactory,
+                            jobServices.GetService<ILogger<ResourcesScrapingJob>>()),
+                    schedulerOptions =>
                     {
-                        schedulerOptions.CronSchedule = scrapeDefinition.Scraping.Schedule;
+                        schedulerOptions.CronSchedule = metricsDeclaration.Metrics[0].Scraping.Schedule;
```

Comment:
This should be the default schedule, not for the first metric, no?

Reply:
I'm not a fan of this and would use a different model for the job scheduling but it's OK for now